ir/state.cpp

```diff
@@ -372,11 +372,14 @@ void State::mkAxioms(State &tgt) {
         for (unsigned i = 0, e = ptr_ins.size(); i != e; ++i) {
           // TODO: needs to take read/read2 as input to control if mem blocks
           // need to be compared
-          expr eq_val = Pointer(mem, ptr_ins[i].value)
-                          .fninput_refined(Pointer(mem2, ptr_ins2[i].value));
+          auto &[ptr_in, is_byval] = ptr_ins[i];
+          auto &[ptr_in2, is_byval2] = ptr_ins2[i];
+          (void)is_byval;
```

Comment:
Assumes that we're only dealing with transformations that don't add byval attributes ; adding such attribute would be interprocedural

Reply:
we should assume as little as possible.
This needs to handle the case where byval is added or removed. It can just set refines to false, for example.